packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -1260,6 +1260,7 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       this.expectContextual(tt._infer);
       const typeParameter = this.startNode();
       typeParameter.name = this.tsParseTypeParameterName();
+      typeParameter.constraint = this.tsEatThenParseType(tt._extends);
```

Comment:
Can you also add this to the type definitions in https://github.com/babel/babel/blob/main/packages/babel-parser/src/types.js, in `@babel/types` and in `@babel/generator`?

Reply:
- I've fixed type definition in `@babel/parser` [3b0b735](https://github.com/babel/babel/pull/14476/commits/3b0b7354d799bb8c10a920da60c5fbf98e2d9014)
- I've added tests in `@babel/generator`.  [a0d32e4](https://github.com/babel/babel/pull/14476/commits/a0d32e436d623cf9a70a16a6b6a7942d663c8238)
- The type for `TSInferType` that is defined in `@babel/types` supports `constraint` field already. So I think we don't need to change `@babel/types` in this PR.